include/mlir-tcp/Dialect/Transforms/Passes.td

```diff
@@ -30,4 +30,11 @@ def VerifyTcpBackendContract : Pass<"torch-verify-tcp-backend-contract", "Module
   let constructor = "mlir::tcp::createVerifyTcpBackendContractPass()";
 }
 
+// \brief This pass decomposes `tensor::ConcatOp` into a `tensor::EmptyOp`
+// and a sequence of `tensor::InsertSliceOp`s.
+def DecomposeTensorConcatOps : Pass<"decompose-tensor-concat-ops", "func::FuncOp"> {
```

Comment:
I'm suspecting we will need to add more such transforms in future. So, can we name this pass something more generic, say, TransformTensorOps, so that we can accommodate more of these if necessary, in future.

Reply:
I've renamed the file to `TransformTensorOpsPass.h/cpp` to keep all tensor op related transforms colocated and minimizing the boilerplate, however chose to keep the pass name for decomposing `tensor.concat` the same for clarity, as I anticipate adding a new transform to the same file but register it as a separate / standalone pass. Does that sound reasonable? We can always reorganize once we uncover more such cases.

Reply:
Personally, I'd prefer having a single pass that covers all the transforms for Tensor dialect ops, because building a pipeline would require knowing all the passes, if we do them separately. 

But we can cross that bridge when we get there, as you said.

Reply:
Renamed the pass to `DecomposeTensorOps` to make room of future tensor decompositions. For something other than a decomposition, I propose adding a separate pass (to the same cpp), and use a function in Pipeline.cpp that  creates a sequence of such "tensor transforms" passes.